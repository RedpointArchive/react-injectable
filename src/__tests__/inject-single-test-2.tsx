import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { InjectSingle } from '../index';

const FooContext = React.createContext<string>("foo");
const BarContext = React.createContext<string>("bar");

interface ComponentProps {
  foo: string;
  bar: string;
  baz: string;
  moo: string;
}

const Component = InjectSingle(
  FooContext,
  "foo",
  InjectSingle(
    BarContext,
    "bar",
    class Component extends React.Component<ComponentProps, {}> {
      public render() {
        return (
          <>
            {this.props.foo}
            {this.props.bar}
            {this.props.baz}
            {this.props.moo}
          </>
        );
      }
    }
  )
);

it('multiple InjectSingle calls inject correctly', () => {
  const renderer = TestRenderer.create(
    <FooContext.Provider value="foo2">
      <BarContext.Provider value="bar2">
        <Component baz="baz" moo="moo" />
      </BarContext.Provider>
    </FooContext.Provider>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});
