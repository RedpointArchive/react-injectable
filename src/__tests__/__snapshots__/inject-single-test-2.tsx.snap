// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple InjectSingle calls inject correctly 1`] = `
Array [
  "foo2",
  "bar2",
  "baz",
  "moo",
]
`;
